jung-visualization/src/main/java/edu/uci/ics/jung/visualization/spatial/SpatialGrid.java

```diff
@@ -206,11 +205,11 @@ protected int getBoxNumber(int[] boxXY) {
    * @param p a location in the coordinate system
    * @return the box number that contains the passed location
    */
-  protected int getBoxNumberFromLocation(Point2D p) {
+  protected int getBoxNumberFromLocation(Point p) {
     int count = 0;
     for (Shape shape : getGrid()) {
       Rectangle2D r = shape.getBounds2D();
-      if (r.contains(p) || r.intersects(p.getX(), p.getY(), 1, 1)) {
+      if (r.contains(p.x, p.y) || r.intersects(p.x, p.y, 1, 1)) {
```

Comment:
is there ever a case where the first clause is false but the second (`r.intersects()`) is true?  the second seems redundant.